### PR TITLE
reuse meta `arrayContentDidChange`

### PR DESCRIPTION
**Pull request by bekzod**

### CODE REVIEW

File: packages/ember-metal/lib/weak_map.js [hunk 1]
```diff
@@ -57,12 +57,12 @@ WeakMap.prototype.get = function(obj) {
   let meta = peekMeta(obj);
   if (meta) {
     let map = meta.readableWeak();
-    if (map) {
-      if (map[this._id] === UNDEFINED) {
+    if (map !== undefined) {
```
Comment (rwjblue) [inline packages/ember-metal/lib/weak_map.js]: This seems unrelated to what this PR title suggests is the goal ("reuse meta arrayContentDidChange"), can you remove that commit and submit separately?

**Review by rwjblue — CHANGES_REQUESTED** [review]

File: packages/ember-metal/lib/weak_map.js (hunk 1 above)
Comment (bekzod) [inline packages/ember-metal/lib/weak_map.js]: Tried to smuggle code sorry 🚓

Comment (rwjblue) [inline packages/ember-metal/lib/weak_map.js]: LOL

**Review by stefanpenner — APPROVED** [review]

File: packages/ember-metal/lib/weak_map.js (hunk 1 above)
Comment (locks) [inline packages/ember-metal/lib/weak_map.js]: bbuussteedd
